### PR TITLE
document applied or removed events implemented

### PR DESCRIPTION
**Pull request by dvoracek-martin**

### CODE REVIEW

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/events/ProofOfWorkEffortsDocumentAppliedEvent.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package ch.admin.seco.alv.service.onlineform.domain.npa.events;
+
+import ch.admin.seco.alv.service.onlineform.core.events.DomainEventType;
+import ch.admin.seco.alv.service.onlineform.domain.npa.ProofOfWorkEfforts;
+
+public class ProofOfWorkEffortsDocumentAppliedEvent extends ProofOfWorkEffortsEvent {
+    public ProofOfWorkEffortsDocumentAppliedEvent(ProofOfWorkEfforts proofOfWorkEfforts) {
+        super(proofOfWorkEfforts, new DomainEventType("ProofOfWorkEffortDocumentApplied"));
```
Comment (emina-str) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/events/ProofOfWorkEffortsDocumentAppliedEvent.java]: you're missing an "s" : `ProofOfWorkEffortsDocumentAppliedEvent` .
Also please add the correct name with a "s" to all other ProofOfWorkEffort**s**Events.

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java [hunk 2]
```diff
@@ -53,4 +55,22 @@ void handleProofOfWorkEffortsSubmittedEvent(ProofOfWorkEffortsSubmittedEvent pro
 				.build());
 	}
 
+	@EventListener
+	void handleProofOfWorkEffortsDocumentAppliedEvent(ProofOfWorkEffortsDocumentAppliedEvent proofOfWorkEffortsDocumentAppliedEvent) {
```
Comment (emina-str) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java]: we don't need that, you can delete it. we only have to listen for this event so we could persist it in the database, and for that we already have a Listener for all DomainEvents -> `EventPersistenceListener.class`, method `persistEvent(DomainEvent<?> event)`. That is enough for now.

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java [hunk 3]
```diff
@@ -53,4 +55,22 @@ void handleProofOfWorkEffortsSubmittedEvent(ProofOfWorkEffortsSubmittedEvent pro
...
+		ProofOfWorkEffortsDto proofOfWorkEffortsDto = this.proofOfWorkEffortsApplicationService.getById(proofOfWorkEffortsDocumentAppliedEvent.getAggregateId());
+		this.proofOfWorkEffortsEventsMessageChannel.send(MessageBuilder
+				.withPayload(proofOfWorkEffortsDto)
+				.setHeader(MessageHeaders.EVENT, proofOfWorkEffortsDocumentAppliedEvent.getDomainEventType().getValue())
+				.build());
+	}
+
+	@EventListener
+	void handleProofOfWorkEffortsDocumentRemovedEvent(ProofOfWorkEffortsDocumentRemovedEvent proofOfWorkEffortsDocumentRemovedEvent) {
```
Comment (emina-str) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java]: same as previous.

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java (hunk 2 above)
Comment (dvoracek-martin) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java]: Done

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/events/ProofOfWorkEffortsDocumentAppliedEvent.java (hunk 1 above)
Comment (dvoracek-martin) [reply domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/events/ProofOfWorkEffortsDocumentAppliedEvent.java]: Done

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java (hunk 3 above)
Comment (pavax) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java]: IMHO: There is no need to send that event to kafka. We can not handle it anywhere. For Example if we'd receive that message in the kofax-exporter we can not tell kofax /avam to delete a document => remove it?

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEfforts.java [hunk 4]
```diff
@@ -66,431 +68,437 @@
...
-	@CollectionTable(name = "WORK_EFFORTS", joinColumns = @JoinColumn(name = "PROOF_OF_WORK_EFFORTS_ID"))
-	private Set<WorkEffort> workEfforts = new HashSet<>();
-
-	@Embedded
-	@AttributeOverride(name = "value", column = @Column(name = "PDF_DOCUMENT_ID"))
-	private DocumentId pdfDocumentId;
-
-	@Enumerated(EnumType.STRING)
-	private Language language;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProofOfWorkEfforts.class);
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEfforts.java]: what happend here with the formatting?

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java (hunk 3 above)
Comment (dvoracek-martin) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java]: Already removed. You and Emina were right that there is no need to process that information afterwards

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEfforts.java (hunk 4 above)
Comment (dvoracek-martin) [reply domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEfforts.java]: The problem is that I and Emina are using spaces, whereas these were probably tabs (by either you or Julien). I guess that it means we should all switch to tabs, right?

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package ch.admin.seco.alv.service.onlineform.infrastructure.messagebroker.events;
 
+import ch.admin.seco.alv.service.onlineform.domain.npa.events.ProofOfWorkEffortsDocumentAppliedEvent;
```
Comment (emina-str) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java]: Remove unused imports.

**Pull request merged**